pkg/controller/machinepool/azureactuator.go

```diff
@@ -95,6 +98,24 @@ func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *
 		workerRole,
 		workerUserDataName,
 	)
+
+	// Use image set for machinepool azure platform instead of the default
+	if image := pool.Spec.Platform.Azure.Image; (machineapi.Image{}) != image {
+		for _, ms := range installerMachineSets {
+			provider := ms.Spec.Template.Spec.ProviderSpec.Value.Object.(*azureprovider.AzureMachineProviderSpec)
+			provider.Image = azureprovider.Image{
```

Comment:
Do we need to worry about which version of OpenShift we are generating the MachineSets for (ie. if we set this for an old (but still supported) version of OpenShift, what happens?)?

Reply:
That is a good question. We probably do.

Reply:
Discussed that extra fields would not produce an error. The managed cluster would ultimately not use the overridden image once machinesets were sent over and possibly error at that point.